activerecord/test/cases/associations/inverse_associations_test.rb

```diff
@@ -625,9 +625,41 @@ def test_child_instance_should_be_shared_with_replaced_via_accessor_parent
     assert_equal f.description, m.face.description, "Description of face should be the same after changes to replaced-parent-owned instance"
   end
 
+  def test_can_set_inverse_on_new_records
+    i = Interest.new
+    m = Man.new
+    assert_nothing_raised { i.man = m }
+  end
+
   def test_trying_to_use_inverses_that_dont_exist_should_raise_an_error
     assert_raise(ActiveRecord::InverseOfAssociationNotFoundError) { Face.first.horrible_man }
   end
+
+  def test_for_new_records_changing_parent_instance_should_change_children_count_in_original_parent
+    m = Man.create
+    i = m.interests.build
+
+    assert_difference("m.interests.size", -1) { i.man = nil }
+  end
+
+  def test_child_can_choose_parent
```

Comment:
For context, this is the main bug this PR tried to address. Since it's been a while and there is a lack of interest. I'm closing this for now.